File: Python/Leetcode Easy/1200MinimumAbsoluteDifference.py
#https://leetcode.com/problems/minimum-absolute-difference/description/
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr = sorted(arr)
        ret = []
        minDistance = float("inf")
        for i in range(1, len(arr)):
            distance = arr[i] - arr[i-1]
            if distance == minDistance: ret.append([arr[i-1], arr[i]])
            if distance < minDistance:
                minDistance = distance
                ret = [[arr[i-1], arr[i]]]
        return ret

File: C++/Leetcode Easy/1200MinimumAbsoluteDifference.cpp
#include <vector>
using namespace std;
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        //[4,2,1,3]
        //1, 2, 3, 4
        vector<vector<int>> res;
        int minDiff = 100000000;
        sort(arr.begin(), arr.end());
        for (int i = 1; i < arr.size(); ++i) {
            if (arr[i]-arr[i-1] < minDiff) {
                minDiff = arr[i]-arr[i-1];
                res = {};
                res.push_back({arr[i-1], arr[i]});
            }
            else if (arr[i]-arr[i-1] == minDiff) {
                res.push_back({arr[i-1], arr[i]});
            }
        }
        return res;
    }
};